reactor-netty-core/src/main/java/reactor/netty/transport/TransportConfig.java

```diff
@@ -311,8 +313,13 @@ protected void loggingHandler(LoggingHandler loggingHandler) {
 		this.loggingHandler = loggingHandler;
 	}
 
-	protected void metricsRecorder(@Nullable Supplier<? extends ChannelMetricsRecorder> metricsRecorder) {
-		this.metricsRecorder = metricsRecorder;
+	protected void metricsRecorder(@Nullable Supplier<? extends ChannelMetricsRecorder> metricsRecorderSupplier) {
+		this.metricsRecorderSupplier = metricsRecorderSupplier;
+		this.metricsRecorder = metricsRecorderSupplier != null ? metricsRecorderSupplier.get() : null;
```

Comment:
here we completely remove the benefit of passing a `Supplier` by eagerly calling it if it is provided. the method is `protected` so we can't really remove it though.

but if we deprecate `Supplier metricsRecorder()` above, we should probably deprecate that one too and document the eager resolving for implementors?

actually, I would probably deprecate that one in priority.

Reply:
also, if we don't return the original `Supplier`, there's no real need for keeping it around in `this.metricsRecorderSupplier` ;)

Reply:
yeah ... true

Reply:
> but if we deprecate Supplier metricsRecorder() above, we should probably deprecate that one too and document the eager resolving for implementors?

On the other hand this is used as a setter for the metrics recorder, so we need just to remove `this.metricsRecorderSupplier`

see here
https://github.com/reactor/reactor-netty/blob/d1f8f10a2dcb2805cfc70c84b1ab2b0ba08d9b1c/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClient.java#L1197
and here
https://github.com/reactor/reactor-netty/blob/d1f8f10a2dcb2805cfc70c84b1ab2b0ba08d9b1c/reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java#L567

Wdyt?

Reply:
yeah, fine by me. we can come back to the `Supplier` part of the API later and refine 👍